README.md

```diff
@@ -68,37 +86,48 @@ Different versions of the example scripts show different ways of describing the
 - `./examples/gobuild` - shows how to use nested invocation to generate LLB for Go package internal dependencies
 
 
-#### Examples
+#### Exploring Dockerfiles
+
+Frontends are components that run inside BuildKit and convert any build definition to LLB. There is a special frontend called gateway (gateway.v0) that allows using any image as a frontend.
 
-##### Starting the buildkitd daemon:
+During development, Dockerfile frontend (dockerfile.v0) is also part of the BuildKit repo. In the future, this will be moved out, and Dockerfiles can be built using an external image.
+
+##### Building a Dockerfile with `buildctl`
 
 ```
-buildkitd --debug --root /var/lib/buildkit
+buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=.
+buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=. --frontend-opt target=foo --frontend-opt build-arg:foo=bar
 ```
 
-The buildkitd daemon suppports two worker backends: OCI (runc) and containerd.
+`--local` exposes local source files from client to the builder. `context` and `dockerfile` are the names Dockerfile frontend looks for build context and Dockerfile location.
 
-By default, the OCI (runc) worker is used.
-You can set `--oci-worker=false --containerd-worker=true` to use the containerd worker.
+##### build-using-dockerfile utility
 
-We are open to add more backends.
-
-##### Building a Dockerfile:
+For people familiar with `docker build` command, there is an example wrapper utility in `./examples/build-using-dockerfile` that allows building Dockerfiles with BuildKit using a syntax familiar to `docker build`.
 
 ```
-buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=.
-buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=. --frontend-opt target=foo --frontend-opt build-arg:foo=bar
-```
+go build -o /usr/local/bin/build-using-dockerfile ./examples/build-using-dockerfile
```

Comment:
Typically this command should be executed as a non-privileged user.

`go build ./examples/build-using-dockerfile && sudo install build-using-dockerfile /usr/local/bin` ?